pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -339,3 +339,13 @@ def test_setitem_boolean_mask(self, mask_type, float_frame):
         expected = df.copy()
         expected.values[np.array(mask)] = np.nan
         tm.assert_frame_equal(result, expected)
+
+    def test_setitem_only_false_indexer_dtype_changed(self):
```

Comment:
can you comment and/or edit the test name to make clear

- is iloc affected?
- does it matter that you have `["b"]` instead of, say, `"b"` or ":"?

comment that the bug is for the value being set being a Series obj

Reply:
is Series affected?

Reply:
Series is not affected. Goes only wrong with ``["b"]`` 

``Iloc`` is actually also affected, but has to be fixed differently probably. In case of ``loc`` we get an empty indexer. In case of ``iloc`` we get an indexer like ``([False],)``, which is not empty per ``is_empty_indexer``. We could adjust ``is_empty_indexer`` or we have to go a different way here